### DISCUSSION TITLE
How to add features to QField Android version?

### DISCUSSION THREAD

**Discussion by sharmapn** [opengisch/QField#5302] [Q&A]
Hi there,

I would like to add a small feature that transforms point data in the QField app - android version.

I just want to know which is best to use. These are some options I came across
- Android Studio
- Visual Studio + Xamarin
- Eclipse
- IntelliJ IDEA
- AIDE
- DroidScript
- Cordova

Also, are there some guides on how to get started with familiarising and modifying the code?

Thanking in advance,
Pankaj

**Comment by m-kuhn** [comment]
To build QField, the [dev doc](https://github.com/opengisch/QField/blob/master/doc/dev.md#android) is the best resource to start. It's not very comprehensive but outlines how you can build the code yourself.
Other than that, the Qt tooling (QtCreator etc) is your friend for IDE.

You probably will want to develop directly on your desktop OS and only cross compile once you are happy.
